src/ansi.rs

```diff
@@ -672,6 +673,70 @@ pub trait Handler {
 
     /// Set mouse cursor icon.
     fn set_mouse_cursor_icon(&mut self, _: CursorIcon) {}
+
+    /// Report current keyboard mode.
+    ///
+    /// ## Caveat
+    ///
+    /// This request must be done successively with the [`identify_terminal`] request, given
+    /// clients rely on the later to identify whether the terminal suppors kitty's keyboard
+    /// protocol.
+    ///
+    /// [`identify_terminal`]: Self::identify_terminal
+    fn report_keyboard_mode(&mut self) {}
+
+    /// Push keyboard mode into the keyboard mode stack.
+    fn push_keyboard_mode(&mut self, _mode: KeyboardMode) {}
+
+    /// Pop the given amount of keyboard modes from the
+    /// keyboard mode stack.
+    fn pop_num_keyboard_modes(&mut self, _to_pop: u16) {}
+
+    /// Set the [`keyboard mode`] using the given [`behavior`].
+    ///
+    /// [`keyboard mode`]: crate::ansi::KeyboardMode
+    /// [`behavior`]: crate::ansi::KeyboardModeApplyBehavior
+    fn set_keyboard_mode(&mut self, _mode: KeyboardMode, _behavior: KeyboardModeApplyBehavior) {}
+}
+
+bitflags! {
+    /// The keyboard mode representing [`kitty keyboard protocol'].
+    ///
+    /// The details on how to build the escape sequences could be found in the
+    /// mentioned specification.
+    ///
```

Comment:
```suggestion
```

I don't think this is necessary, especially because the mode isn't really directly related to the escape sequence.